containers.go

```diff
@@ -89,8 +89,11 @@ func (c *ContainerDefinition) DockerMounts() ([]docker.HostMount, error) {
 				src = filepath.Join(c.LocalWorkDir, src)
 			}
 			// TODO do the same for dst?
-			if err := os.MkdirAll(src, 0777); err != nil {
-				return []docker.HostMount{}, fmt.Errorf("Couldn't make source dir %s: %v", src, err)
+			// os.Stat prevents /var/run/docker.sock from erroring out as "not a directory"
```

Comment:
@zombiezen and @jamesnaftel, #7 removed the MkDirIfNeeded function, that did this. Maybe trying to revert some bits and plumb it all around would be helpful :man_shrugging: ?

Reply:
I don't think reverting #7 is the right call. My issue with this code is that it drops any errors that aren't non-existence errors on the ground.

Can we take a step back and examine why we're calling `os.MkdirAll` here? Is it actually a safe assumption that all of the mounts are directories (seems like no)? Is there a better behavior we can enforce here?

Reply:
Good point

Reply:
I'm ok with stepping back and learning.  I was simply solving the immediate problem that staging was broken.  I'm trying to view the SquashImage behavior in staging so we can wrap up that issue.  I definitely do not understand all of the mount use cases but it would be good for some to document them.

Reply:
@jamesnaftel I'm totally onboard with your movement forward and getting you unblocked; this PR was the right one. My pushback is on @beholders-eye's idea to revert the Narwhal v0.2 changes: rolling forward seems more correct.

It's also probably not the most important thing at this moment to figure out the right answer for this part. I'd be fine with us opening a chore ticket and putting it on backlog for investigation.

Reply:
I think I didn't choose my words wisely, by revert I've meant to cherry pick some logic back from the dead code. But you're right @zombiezen, let's move forward. Thanks!